app/generateSMTScriptProp.ts

```diff
@@ -0,0 +1,7 @@
+interface Smt { smt2: string; propositions: Set<string>; }
+
+export function generateSMTScriptProp(userSmt: Smt, sysSmt: Smt): string {
```

Comment:
I'd rename `userSmt` to something super human readable. What would you call it when explaining it to someone in an elevator? Is it important that the first of these SMT's come from the user and the second from the sys? Is there a name for this function that clearly communicates what you're trying to achieve (once again, if you were explaining it to someone in an elevator)?